bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/item/EnrichedItemDTOMapper.java

```diff
@@ -64,7 +65,9 @@ private static EnrichedItemDTO map(Item item, ItemDTO itemDTO, URI uri, boolean
             if (drillDown) {
                 Collection<EnrichedItemDTO> members = new LinkedHashSet<>();
                 for (Item member : groupItem.getMembers()) {
-                    members.add(map(member, drillDown, uri, locale));
+                    if (itemTypeFilter == null || itemTypeFilter.equals(member.getType())) {
```

Comment:
If there is an itemTypeFilter, you will never drill down any further now, is this expected?

Reply:
Yes, that is correct.
Your use case was to get all Location items inside a group. I thinkmy code matches your use case.

Reply:
The method is called with drillDown=true, which means that it traverses the tree. I'd therefore say it is not the generally expected result (while we can discuss which way we want it for the mapview widget).